UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs

```diff
@@ -331,16 +332,31 @@ public static List<PushPull> GetPushableAt(Vector3Int worldOrigin, Vector2Int di
 	{
 		Vector3Int worldTarget = worldOrigin + dir.To3Int();
 		List<PushPull> result = new List<PushPull>();
+
 		foreach ( PushPull pushPull in GetAt<PushPull>(worldTarget, isServer) )
 		{
+			PushPull pushable = pushPull;
+
 			if ( pushPull && pushPull.gameObject != pusher && (isServer ? pushPull.IsSolidServer : pushPull.IsSolidClient) )
 			{
+				// If the PushPull parent object is a player, and that player is buckled, we should use the pushPull object that the player is buckled to.
```

Comment:
in this section, I think you can use GetComponent in both places instead of GetComponentInParent